### PR TITLE
Introduce graphs as Oscar objects

### LINKED ISSUE #663: Polymake lattice_points not in the documentation?
It seems like the Polymake function lattice_points(P::Polyhedron) exists but is not mentioned in the documentation. Is that on purpose? It would be nice if this could be added.

### PR DESCRIPTION
**Pull request by lkastner**
We interface the polymake graphs from Oscar toplevel. The implementation uses the `LightGraphs` as blueprint.

Alongside we introduce some changes for accessing polymake multiple subobjects (e.g. a linear program is a multiple subobject of a polyhedron). Previously only the first such subobject could be accessed. (on the corresponding branch in libpolymake_julia)

### CODE REVIEW

**Comment by thofma** [comment]
`ne` and `nv` ... I guess these names come from `LightGraphs`?

**Comment by lkastner** [comment]
> `ne` and `nv` ... I guess these names come from `LightGraphs`?

Yes, unfortunately. @benlorenz is suggesting to encapsulate this in a submodule.

**Comment by lkastner** [comment]
So I tried making this into a submodule. One can now do `using Oscar.Graphs`. I read up on documentation, but I was unable to figure out whether this is the legal way to do this.

**Comment by fieker** [comment]
On Fri, Sep 03, 2021 at 06:01:53AM -0700, Lars Kastner wrote:
> So I tried making this into a submodule. One can now do `using Oscar.Graphs`. I read up on documentation, but I was unable to figure out whether this is the legal way to do this.
> 
we're using them in various places in Hecke...
you can do using .Graphs inside of Oscar

**Comment by thofma** [comment]
How far is this? There are some applications where would like to compute automorphism groups.

**Comment by benlorenz** [comment]
I am currently waiting for a bump of `polymake_jll` to be merged, once that is done I can create the `libpolymake_julia_jll` releases, and then finally the Polymake.jl release for this.
All the code should be done and was already tested locally.

**Comment by thofma** [comment]
Thanks for the update!

**Comment by thofma** [comment]
Nice, but the doctests are failing. The `nv` calls need a prefix.

**Comment by lkastner** [comment]
> Nice, but the doctests are failing. The `nv` calls need a prefix.

It looks like everything needs a prefix, but I don't understand why. Locally it works for me and I had added
```
DocMeta.setdocmeta!(Oscar.Graphs, :DocTestSetup, :(using Oscar; using Oscar.Graphs); recursive = true)
```
in the `make_work.jl` to make it work. Any hints?

File: docs/make_work.jl [hunk 1]
```diff
@@ -63,6 +63,7 @@ function doit(Oscar::Module; strict::Bool = true, local_build::Bool = false, doc
 
     DocMeta.setdocmeta!(Oscar, :DocTestSetup, :(using Oscar); recursive = true)
     DocMeta.setdocmeta!(Oscar.Hecke, :DocTestSetup, :(using Hecke); recursive = true)
+    DocMeta.setdocmeta!(Oscar.Graphs, :DocTestSetup, :(using Oscar; using Oscar.Graphs); recursive = true)
```
Comment (thofma) [inline docs/make_work.jl:66]:
```suggestion
    DocMeta.setdocmeta!(Oscar, :DocTestSetup, :(using Oscar; using Oscar.Graphs); recursive = true)
```

Comment (thofma) [reply docs/make_work.jl:66]: Let's try this.

File: src/Polytopes/Cone/properties.jl [hunk 2]
```diff
@@ -161,7 +161,7 @@ isfulldimensional(C::Cone) = pm_cone(C).FULL_DIM
 ## Points properties
 ###############################################################################
 
-"""
+@doc Markdown.doc"""
```
Comment (fingolfin) [inline src/Polytopes/Cone/properties.jl:164]: There seems to be a lot in this PR which doesn't have anything to do with graphs?

File: src/Polytopes/Graphs.jl [hunk 3]
```diff
@@ -0,0 +1,659 @@
...
+```jldoctest
+julia> c = cube(3);
+
+julia> g = edgegraph(c);
+
+julia> nv(g)
+8
+```
+"""
+function nv(g::Graph{T}) where {T <: Union{Directed, Undirected}}
```
Comment (fingolfin) [inline src/Polytopes/Graphs.jl:313]: Ugh... `nv`, really? LightGraphs, whyyyyy??? I don't get this desire to shorten identifiers so much, esp. for exported stuff. If everybody does it, composing code from different sources becomes ever harder. Urgh.

And `ne` is even worse, it stands for `not equal` in a bunch of places (TeX, many assembler dialects). 

How about we give it a sane name, e.g `nverts`, and similar `ne` -> `nedges`. And if someone for some reason really needs `LightGraphs.jl` compatibility (is that actually a thing?), then adding alias methods is a matter of adding 2 short lines of code.

Comment (thofma) [reply src/Polytopes/Graphs.jl:313]: We agreed above to add those awful names, but hide them in the `Graphs` submodule. The proper names would be `num_edges` I guess (to be in line with the other `num_*` functions)?

**Comment by thofma** [comment]
@fingolfin, are the issues you raised blocking? If not I would like to merge this soonish.

File: src/Polytopes/Graphs.jl (hunk 3 above)
Comment (fingolfin) [reply src/Polytopes/Graphs.jl:313]: OK if it is hidden, that's better.

The naming issue is less clear to me. Loading Oscar, I see only a single `num_` function, `num_coeff`. But I see a bunch of `number_` functions (some from GAP/groups, e.g. `number_conjugacy_classes`; others from Polymake/Polytopes, e.g. `number_of_factors`), and of course also `n*` functions (e.g. `nrows`, `ncols`, `nvertices`, `nfacets`); of course Julia itself uses `n*` e.g. `ndims`, `nfields`, ...). Perhaps another issue should be opened to settle *that* bit of (in)consistency... *sigh*

**Review by fingolfin — APPROVED** [review]

File: src/Polytopes/Graphs.jl (hunk 3 above)
Comment (thofma) [reply src/Polytopes/Graphs.jl:313]: Ah, you are right. Sad.

**Pull request merged**